lib/system.nim

```diff
@@ -1144,46 +1158,53 @@ proc `<`*(x, y: int64): bool {.magic: "LtI", noSideEffect.}
 type
   IntMax32 = int|int8|int16|int32
 
-proc `+%`*(x, y: IntMax32): IntMax32 {.magic: "AddU", noSideEffect.}
-proc `+%`*(x, y: int64): int64 {.magic: "AddU", noSideEffect.}
+proc `+%`*(x, y: IntMax32): IntMax32 {.magic: "AddU", noSideEffect, deprecated.}
```

Comment:
Gah, no, only deprecate ``ze`` and ``toU`` procs please.

Reply:
And concrete reason why you want to keep these ugly and unnecessary operators alive? I am sure they won't be removed for a long time, even if we deprecate them now, since you really wrote a lot of old code with it. But that is not really a reason to me to not deprecate them since I really don't want anybody to write new code with them.

Reply:
> even if we deprecate them now, since you really wrote a lot of old code with it.

I don't want to ever fix it, that's why. It's also simply a different CPU opcode, it's an addition (subtraction etc.) with different overflow behavior.

> since I really don't want anybody to write new code with them.

And I like them, I don't mind new code using them.

Reply:
I don't have strong opinions for or against.

I use them for low-level pointer arithmetics and also to avoid overflows at specific places when implementing a big int library for example.

I don't think they belong to system however, maybe in `sugar` or an `unsigned` module?

Reply:
@mratsim why don't you use unsigned integers for that use case?

Reply:
@Araq btw your feedback has been applied.